src/sentry/metrics/minimetrics.py

```diff
@@ -191,13 +194,42 @@ def _flush(self) -> None:
 
             time.sleep(2.0)
 
-    def _emit(self, extracted_metrics: Any) -> Any:
+    @staticmethod
+    def _emit(extracted_metrics: Any) -> Any:
         # In order to avoid an infinite recursion for metrics, we want to use a thread local variable that will
         # signal the downstream calls to only propagate the metric to the primary backend, otherwise if propagated to
         # minimetrics, it will cause unbounded recursion.
         thread_local.in_minimetrics = True
-        # We want to emit a metric on how many metrics we would technically emit if we were to use minimetrics.
-        metrics.incr("minimetrics.emit", amount=len(extracted_metrics))
+
+        # We obtain the counts for each metric type, since we want to know how many by type we have.
+        counts_by_type: Dict[str, float] = defaultdict(lambda: 0)
```

Comment:
I would prefer we use a regular dict with `setdefault`. The issue with defaultdicts is that they sometimes make their way to places that assume that they get a key error on missing items and I want to avoid this type of bug.